apps/opentelemetry_api_experimental/lib/open_telemetry/counter.ex

```diff
@@ -0,0 +1,26 @@
+defmodule OpenTelemetryAPIExperimental.Counter do
+  @moduledoc """
+
+  """
+
+  defmacro create(name, value_type, opts) do
+    quote bind_quoted: [name: name, value_type: value_type, opts: opts] do
+      :otel_meter.create_counter(
+        :opentelemetry_experimental.get_meter(__MODULE__),
+        name,
+        value_type,
+        opts
+      )
+    end
+  end
+
+  defmacro add(name, number, attributes) do
+    quote bind_quoted: [name: name, number: number, attributes: attributes] do
+      :otel_counter.add(
+        :otel_meter.lookup_instrument(:opentelemetry_experimental.get_meter(__MODULE__), name),
+        number,
+        attributes
+      )
+    end
+  end
+end
```

Comment:
```suggestion
defmodule OpenTelemetryAPIExperimental.Counter do
  @moduledoc """

  """

  defmacro create_counter(name, value_type, opts) do
    quote bind_quoted: [name: name, value_type: value_type, opts: opts] do
      :otel_meter.create_counter(
        :opentelemetry_experimental.get_meter(__MODULE__),
        name,
        value_type,
        opts
      )
    end
  end

  defmacro add_counter(name, number, attributes) do
    quote bind_quoted: [name: name, number: number, attributes: attributes] do
      :otel_counter.add(
        :otel_meter.lookup_instrument(:opentelemetry_experimental.get_meter(__MODULE__), name),
        number,
        attributes
      )
    end
  end
end
```

I prefer when the functions are self-sufficient and treat the module as a namespace that I could often ignore together.

:2cents: but 🤷🏻 whatever works, you get used to it.

Reply:
Hm, interesting point. I guess we should go with what is most common in Elixir libraries. I'll go look at some like prometheus.ex and telemetry metrics.

Reply:
Except for functions like `new` where you create creating structs out of the module (so it does matter a lot the name of the module), I think it is a typical pattern for the most part.

The biggest concern is when you require to `import` functions since the more generic you go, the harder it is because things are too generic. Or when you have a macro that also expands things in the scope.

I tend to avoid all together functions that are incapable of being self-sufficient. At some point, collisions will happen, and in the worst case, I can not even control it because it is in a 3rd party package ... so it becomes tricky to figure out the right solution.

Also, it is inlined with what the erlang package is doing. So it would be nice to avoid that much translation.

Reply:
I'm going to merge this but not consider this issue settled. I'd like more input before making a decision but don't want to hold up this PR , since its not being released, and even if it was is still experimental :)

Reply:
🚀 🚀 🚀